src/explorepy/explore.py

```diff
@@ -254,8 +257,15 @@ def device_info_callback(packet):
                                                             adc_mask=self.stream_processor.device_info['adc_mask'],
                                                             do_overwrite=do_overwrite)
                 self.recorders['marker'] = self.recorders['exg']
+                self.recorders['meta'] = create_meta_recorder(filename=meta_out_file,
```

Comment:
`meta_out_file` is not redefined in the callback. Python will use the variable defined in the outer space which causes overwriting of the previous meta data file.